src/ext/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSource.java

```diff
@@ -458,9 +458,9 @@ private boolean handleTripPatternUpdate(
    * Snapshot timetable is used as source if initialised, trip patterns scheduled timetable if not.
    */
   private Timetable getCurrentTimetable(TripPattern tripPattern, LocalDate serviceDate) {
-    TimetableSnapshot timetableSnapshot = getTimetableSnapshot();
+    TimetableSnapshot timetableSnapshot = snapshot;
     if (timetableSnapshot != null) {
-      return getTimetableSnapshot().resolve(tripPattern, serviceDate);
+      return timetableSnapshot.resolve(tripPattern, serviceDate);
```

Comment:
`timetableSnapshot` can be inlined

Reply:
I would rather keep it this way. It should not happen the way it is implemented right now but potentially snapshot can change as we proceed with updates. I would rather have it assigned to separate variable so that we are sure we operate on same instance all the time.

Reply:
Sure, it should never set back to null after an initial update. I think we want to change the logic, so that we make sure, it can't be updated back to null, but can live with this now.